README.md

```diff
@@ -78,9 +123,135 @@ You can see the basic dummy app here:
+You would then be able to create a field
+with an autocomplete widget as follows:
+
+```
+{{fm-field
+  widget='autocomplete'
+  value=value
+  widgetAttrs=(hash
+    choices=autocompleteChoices
+    placeholder=placeholder
+  )
+}}
+```
+
+The `fm-field` component will take care of adding labels, errors and
+formatting classes around your autocomplete widget.  Meanwhile your
+autocomplete widget can focus solely on adding the extra autocomplete
+functionality you want.
+
+You can create custom form widgets easily.  Simply create a new
+component named as `fm-widgets/<your widget name>`. A widget will
+receive the following attrs:
+
+- `value`: the current value of the input.
+
+- `action` (optional): action to call with a new value chosen through
+  user interaction with the widget. If you do not want to support
+  actions-up/data-down design you can modify `value` directly and ignore
+  `action`.
+
+- `widgetAttrs`: an Ember.Object which is used as a vessel to pass any
+  custom values for your widget.  For instance a select, widget would
+  look for the choices, labelField and valueField properties in the
+  `widgetAttrs` property.
+
+- `onUserInteraction`: your custom widget needs to call this action
+  whenever it is manipulated. This lets `fm-field` know when it should
+  show errors on the form field.
+
+- `classNames`: allows the `fm-field's` display component to add certain
+  standard css classes.
+
+#### Displays
+
+Displays let you customize the aesthetics of a form field.  For
+instance, you may want to create a display where the fields label and
+inputs appear in rows.  You could do this by creating a component
+called `fm-displays/horizontal`. Your
+`templates/components/fm-displays/horizontal.hbs` might look like this:
+
+```
+<div class="{{styles.wrapperClass}}">
+  <label for="{{forAttribute}}" class="col-sm-2 {{styles.labelClass}}">
+    {{label}}
+  </label>
+  <div class="col-sm-10">
+    {{yield inputClasses}}
+  </div>
+
+  {{#if visibleErrors.length}}
+    <span class="{{styles.errortextClass}}">
+      {{visibleErrors.firstObject}}
+    </span>
+  {{/if}}
+
+  {{#if helptext}}
+    <span class="{{styles.helptextClass}}">
+      {{helptext}}
+    </span>
+  {{/if}}
+</div>
+```
+
+This widget will be included wherever you place the `yield` statement.
+The `inputClasses` attribute is passed to the `yield` statement to
+specify classes that should be applied to the widget itself.
+
+You can now use this display anywhere in your application by simply specifying
+the layout when creating an `fm-field`:
+
+```
+{{fm-field
+  display='horizontal'
+  widget='input'
+  value=myValue
+  errors=errors
+  placholder=placehodler
+}}
+```
+
+Note how we defer to `styles` property for class names where possible.
+This ties in the `fmConfig` service which provides easy configuration of
+form classes on a global basis.
+
+The best place to start for creating a display is to look at
+`fm-displays/default.hbs`.
+
 ### Errors
```

Comment:
Would love to add a couple notes on how people should approach client side vs server side, even if it's beyond the scope of this library